src/evaluator.ml

```diff
@@ -1,31 +1,146 @@
 open Ast
+open Common
 
 
-let rec check_if_dec_in_db (Clause (h, b), db) =
-    match db with
-    | [] -> false
-    | ((Clause (h1, b1)) :: xs) ->
-        if h = h1
-        then true
-        else check_if_dec_in_db (Clause (h,b) ,xs)
+let (fresh, reset) =
+  let nxt = ref 0 in
```

Comment:
What's a `ref` and what does `:=` do?

Reply:
ref is a makes it a reference type (like pointers).